### PR TITLE
Resolving issue #669 - bug in watcher

### PR DESCRIPTION
**Pull request by samifouad**
testing out github desktop for the first time. sorry if this is a completely messed up pull request.

### CODE REVIEW

**Comment by samifouad** [comment]
**Damn it**. It must not have fast forwarded my fork before the pull request. Sorry. 

I only actually want to submit a change to client.js and not the already resolve tree-view.js issue.

File: lib/client.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ export default (function INIT() {
 
           const dir = self.getProjectPath();
 
-          const watchDataFormatted = watchData.map(watch => Path.resolve(dir, watch));
+          const watchDataFormatted = watchData.map(watch => Path.resolve(dir+watch));
```
Comment (jpxd) [inline lib/client.js:53]: This is strange.. it should not change the behavior... 
I'll test this, but can you elaborate on why Path.resolve did not work the way it was intended?
See https://nodejs.org/api/path.html#path_path_resolve_paths

**Comment by samifouad** [comment]
I don't understand it either. I actually re-wrote the entire implementation of watcher and it was working, then I kind of started from scratch again from the beginning to compare mine to the current version. The very first thing I coded differently was that line and after a test, it worked.

Without the change, for reasons I did not investigate the array never gets any data in the current code.

It feels weird that after re-writing the entire section I just submit a single character change, but sometimes that's how debugging goes I guess. lol

**Comment by samifouad** [comment]
When I did my re-write, I wrote in a file deletion and file rename feature to watcher. I completely forgot to save my code but it shouldn't be too difficult to add again if that's something that would be nice to have.

Before I submit those feature additions, I figured I would ask someone about the changes.

Basically right now, chokidar will send an event when a file changes and the code deals with the change accordingly, but when chokidar detects a rename event or a deletion event, it is not handled.  So with a few more methods and some tweaking, I could add those features. It was working quite nicely during my test build.

Also, that kind of got me thinking, should keeping the ENTIRE local directory be kept in sync via chokidar? That's honestly not too difficult to implement if an md5 or sha of the files are kept some where and it could also use modified time info as a way to detect what file should be overwritten. So the one part about that sort of a system that would present a challenge would be gracefully dealing with the conflicts. The users could always be given a config variable such as "Remote-FTP.syncConflict" with options as either "1" for always keep new files and "2" for prompt for every conflict.

I'm not familiar enough with Atom to make a good way to present a conflict and then have the user view and take action. Adding this feature would definitely be a good learning experience for dealing with Atom views.

**Comment by samifouad** [comment]
I inadvertently bundled this change to my more recent pull request, #697 

So I'll close this one

**Pull request closed without merging**